ITEvents/Screens/EventCollectionViewCommon.swift

```diff
@@ -0,0 +1,112 @@
+        let event = events[indexPath.row]
+        if isListLayoutSelected {
+            let cell: ListCollectionViewCell = collectionView.dequeueReusableCell(for: indexPath)
+            cell.setup(with: event)
+            return cell
+        }
+        let cell: GridCollectionViewCell = collectionView.dequeueReusableCell(for: indexPath)
+        cell.setup(with: event)
+        return cell
+    }
+    
+    func collectionView(_ collectionView: UICollectionView,
+                        viewForSupplementaryElementOfKind kind: String,
+                        at indexPath: IndexPath) -> UICollectionReusableView {
+        switch kind {
+        case UICollectionElementKindSectionFooter:
+            let footer = collectionView.dequeueReusableSupplementaryView(ofKind: UICollectionElementKindSectionFooter,
+                                                                         withReuseIdentifier: "Footer",
+                                                                         for: indexPath) as! EventsCollectionViewFooter
+            if loadInProgress {
+                footer.showFooter()
+            } else {
+                footer.hideFooter()
+            }
+            return footer
+        default:
+            assert(false, "Unexpected element kind")
+        }
+    }
+    
+    func collectionView(_ collectionView: UICollectionView,
+                        layout collectionViewLayout: UICollectionViewLayout,
+                        referenceSizeForFooterInSection section: Int) -> CGSize {
+        return loadInProgress
+            ? CGSize(width: viewWidth, height: 50)
+            : CGSize.zero
+    }
+    
+    func collectionView(_ collectionView: UICollectionView,
+                        layout collectionViewLayout: UICollectionViewLayout,
+                        sizeForItemAt indexPath: IndexPath) -> CGSize {
+        return isListLayoutSelected
+            ? CGSize(width: viewWidth - 20, height: 80)
+            : CGSize(width: (viewWidth - 40) / 2, height: 180)
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        collectionView.isUserInteractionEnabled = false
+        selectedEventAction?(indexPath)
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, willDisplay cell: UICollectionViewCell, forItemAt indexPath: IndexPath) {
+        lastCellWillDisplayAction?(indexPath)
+    }
+    
+    func setEvents(_ newEvents: [EventCollectionCellViewModel]) {
+        self.events = newEvents
+    }
+    
+    func toggleLoadingProgressState(_ value: Bool) {
+        loadInProgress = value
+    }
+    
+    func toggleLayout(value isListLayout: Bool) {
+        if isListLayoutSelected == isListLayout { return }
+        isListLayoutSelected = isListLayout
+    }
+    
+    func getLoadingState() -> Bool {
+        return loadInProgress
+    }
+    
+    //Do not know how to name it
+    func getLayoutState() -> Bool {
```

Comment:
Lets remove the  ```getLayoutState()```  method and instead use the well named field ```isListLayoutSelected```, we just make it to be editable only inside the class:
```swift
// I specify that only the setter is private!
public private(set) var isListLayoutSelected = true
```

So we will be able to use it outside of this class like this:
```swift
if (eventCollectionViewCommon.isListLayoutSelected) {
   ...
}
```

And it should not allow us to change the field like this:
```swift
// should be compile error here
eventCollectionViewCommon.isListLayoutSelected = true; 
```